Revised Introduction (#60)
greenelab/text_mined_hetnet_manuscript: content/02.introduction.md

## Introduction

Knowledge bases are essential resources that hold complex structured and unstructured information. 

> These resources have been used in important tasks such as constructing networks for drug repurposing discovery [@doi:10.1371/journal.pone.0084912; @doi:10.1101/385617; @doi:10.7554/eLife.26726] or serve as a source of training labels for text mining systems [@doi:10.3115/1690219.1690287; @doi:10.1101/444398; @doi:10.1186/s12859-019-2873-7]. 

Reviewer: ```suggestion
These resources have been used to construct networks for drug repurposing discovery [@doi:10.1371/journal.pone.0084912; @doi:10.1101/385617; @doi:10.7554/eLife.26726] or as a source of training labels for text mining systems [@doi:10.3115/1690219.1690287; @doi:10.1101/444398; @doi:10.1186/s12859-019-2873-7]. 
```

> Populating knowledge bases often requires highly trained scientists to read biomedical literature and summarize the results, a process known as manual curation [@doi:10.1093/bib/bbn043].

Reviewer: ```suggestion
Populating knowledge bases often requires highly trained scientists to read biomedical literature and summarize the results through manual curation [@doi:10.1093/bib/bbn043].
```

Supervised systems depend on machine learning classifiers to predict the existence of a relationship using biomedical text as input.
These classifiers can range from linear methods such as support vector machines [@doi:10.1371/journal.pone.0200699; @doi:10.1093/bioinformatics/btw503] to deep learning [@doi:doi:10.1093/database/bay102; @doi:10.1016/j.neunet.2014.09.003; @arxiv:1904.02181; @arxiv:1901.08746; @arxiv:1706.03762; @doi:10.1093/database/bay060], which all require access to well-annotated datasets.
Typically, these datasets are usually constructed via manual curation by individual scientists [@doi:10.1186/s12859-015-0472-9; @doi:10.1016/j.jbi.2012.04.004; @doi:10.1016/j.artmed.2004.07.016; @doi:10.1186/1471-2105-8-50; @doi:10.1093/bioinformatics/btl616]  or through community-based efforts [@doi:10.1093/database/baw068; @biocreative-chemprot; @doi:10.1186/1471-2105-9-S3-S6].
Often, these datasets are well annotated but are modest in size, making model training hard as these algorithms become increasingly complex.

> Distant supervision is a paradigm that quickly sidesteps manual curation dependence to generate large training datasets.

Reviewer: ```suggestion
Distant supervision is a paradigm that quickly sidesteps manual curation to generate large training datasets.
```

This technique assumes that positive examples have been previously established in selected databases, implying that the corresponding sentences or data points are also positive [@doi:10.3115/1690219.1690287].
The central problem with this technique is that generated labels are often of low quality, resulting in many false positives [@jiang2018revisiting].
Despite this caveat there have been notable effort using this technique [@doi:10.1093/bioinformatics/btv476; @doi:10.1007/978-981-13-2354-6_39; @doi:10.1093/bioinformatics/btz490].

> Data programming is one proposed solution to amend the false positives problem caused by distant supervision.

Reviewer: ```suggestion
Data programming is one proposed solution to amend the false positive problem in distant supervision.
```

> This medium combines labels obtained from distant supervision with simple rules and heuristics written as small programs called label functions  [@arxiv:1605.07723].

Reviewer: ```suggestion
This strategy combines labels obtained from distant supervision with simple rules and heuristics written as small programs called label functions  [@arxiv:1605.07723].
```

These outputs are consolidated via a noise-aware model to produce training labels for large datasets.
Using this paradigm can dramatically reduce the time required to obtain sufficient training data; however, writing a helpful label function requires substantial time and error analysis.
This dependency makes constructing a knowledge base with a myriad of heterogenous relationships nearly impossible as tens or hundreds of label functions are necessary per relationship type.  

This paper seeks to accelerate the label function creation process by measuring how label functions can be reused across different relationship types.

> We hypothesize that sentences describing one relationship type may share linguistic features such as keywords or sentence structure with sentences describing other relationship types.

Reviewer: ```suggestion
We hypothesized that sentences describing one relationship type might share linguistic features such as keywords or sentence structure with sentences describing other relationship types.
```

> If this hypothesis holds, one can drastically reduce the time needed to build a relation extractor system and swiftly populate large databases like Hetionet v1.

Reviewer: ```suggestion
If this hypothesis were to, one could drastically reduce the time needed to build a relation extractor system and swiftly populate large databases like Hetionet v1.
```

We conducted a series of experiments to estimate how label function reuse enhances performance over distant supervision alone.

> We focus on relationships that indicate similar types of physical interactions (i.e., gene-binds-gene and compound-binds-gene) along with two mutually exclusive types (i.e., disease-associates-gene and compound-treats-disease).

Reviewer: ```suggestion
We focused on relationships that indicated similar types of physical interactions (i.e., gene-binds-gene and compound-binds-gene) and two more distinct types (i.e., disease-associates-gene and compound-treats-disease).
```
